tools/RAiDER/interpolator.py

```diff
@@ -91,21 +93,19 @@ def interpVector(vec, Nx):
 
 
 def fillna3D(array, axis=-1):
-    '''
-    Fcn to fill in NaNs in a 3D array by interpolating over one axis only
-    '''
-    # Need to handle each axis
-    narr = np.moveaxis(array, axis, -1)
-    shape = narr.shape
-    y = narr.flatten()
-
-    test_nan = np.isnan(y)
-    def finder(z): return z.nonzero()[0]
-
-    try:
-        y[test_nan] = np.interp(finder(test_nan), finder(~test_nan), y[~test_nan])
-        newy = np.reshape(y, shape)
-        final = np.moveaxis(newy, -1, axis)
-        return final
-    except ValueError:
-        return array
+    
+    XX, YY, temp2 = array.shape
+    
+    if ((axis == 0)|(axis == 1)):
+        temp_pd = pd.DataFrame()
+        temp_pd = temp_pd.interpolate(method='linear',axis=axis,limit_direction='both')
+        array = temp_pd.to_numpy()
+    elif (axis == -1):
+        for yy in range(YY):
+            temp_pd = pd.DataFrame(data=array[:,yy,:], dtype=np.float32)
+            temp_pd = temp_pd.interpolate(method='linear',axis=1,limit_direction='both')
+            array[:,yy,:] = temp_pd.to_numpy()
+    else:
+        raise Exception("Axis out of the array shape")
+    
+    return array
```

Comment:
You can simplify this using np.moveaxis and reshape:

```suggestion
    narr = np.moveaxis(array,axis,-1)
    nars = narr.reshape((np.prod(narr.shape[:-1]),) +(narr.shape[-1],))
    dfd = pd.DataFrame(data=nars).interpolate(axis=1,limit_direction='both')
    out = dfd.values.reshape(array.shape)
    return np.moveaxis(out,-1,axis)
```

Reply:
Great. That is so concise! Really like the way you rewrote. I revised this function as well as the unit test per your suggestion. The CI check ran successfully for the interpolator unit test. Still failed at test_scenario_1 and test_scenario_2 for the reasons mentioned in my email. If you agree, I can re-create those true-value files, which will let the CI check go through and conclude this PR.